stages/pod-fast.yaml

```diff
@@ -0,0 +1,124 @@
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+      - key: '.metadata.deletionTimestamp'
+        operator: 'DoesNotExist'
+      - key: '.status.podIP'
+        operator: 'DoesNotExist'
+  next:
+    statusTemplate: |
+      {{ $now := Now }}
+      conditions:
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: Initialized
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: Ready
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: ContainersReady
+        {{ range .spec.readinessGates }}
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: {{ .conditionType }}
+        {{ end }}
+
+      containerStatuses:
+        {{ range .spec.containers }}
+        - image: {{ .image }}
+          name: {{ .name }}
+          ready: true
+          restartCount: 0
+          state:
+            running:
+              startedAt: {{ $now }}
+        {{ end }}
+
+      initContainerStatuses:
+        {{ range .spec.initContainers }}
+        - image: {{ .image }}
+          name: {{ .name }}
+          ready: true
+          restartCount: 0
+          state:
+            terminated:
+              exitCode: 0
+              finishedAt: {{ $now }}
+              reason: Completed
+              startedAt: {{ $now }}
+        {{ end }}
+
+      {{ with .status }}
+      hostIP: {{ with .hostIP }} {{ . }} {{ else }} {{ NodeIP }} {{ end }}
+      podIP: {{ with .podIP }} {{ . }} {{ else }} {{ PodIP }} {{ end }}
+      {{ end }}
+
+      phase: Running
+      startTime: {{ $now }}
+---
+kind: Stage
+apiVersion: kwok.x-k8s.io/v1alpha1
+metadata:
+  name: pod-completed-for-job
+spec:
+  resourceRef:
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+      - key: '.metadata.deletionTimestamp'
+        operator: 'DoesNotExist'
+      - key: '.status.phase'
+        operator: 'In'
+        values:
+          - 'Running'
+      - key: '.metadata.ownerReferences.[].kind'
+        operator: 'In'
+        values:
+          - 'Job'
+  next:
+    delete: false
+    statusTemplate: |
+      {{ $now := Now }}
+      {{ $root := . }}
+      containerStatuses:
+        {{ range $index, $item := .spec.containers }}
+        {{ $origin := index $root.status.containerStatuses $index }}
+        - image: {{ $item.image }}
+          name: {{ $item.name }}
+          ready: true
+          restartCount: 0
+          started: true
+          state:
+            terminated:
+              exitCode: 0
+              finishedAt: '{{ $now }}'
+              reason: Completed
+              startedAt: '{{ $now }}'
+        {{ end }}
+      phase: Succeeded
+---
+kind: Stage
+apiVersion: kwok.x-k8s.io/v1alpha1
+metadata:
+  name: pod-delete
+spec:
+  resourceRef:
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+      - key: '.metadata.deletionTimestamp'
+        operator: 'Exists'
+  next:
```

Comment:
I assume `delay` is a general directive so I can craft one here to simulate terminationSeconds?

Reply:
Yeah, `delay` is a generic option that can be added to any `Stage`, but `pod-fast` and node-fast are compatible with the original logic, so there is no `delay`.


Something like this can be done by taking a random value between 1 second and `.metadata.deletionTimestamp` as the delay
``` yaml
  delay:
    durationSeconds: 1
    jitterDurationSecondsFrom:
      expressionFrom: '.metadata.deletionTimestamp'
```